src/components/MoneyRequestConfirmationList.js

```diff
@@ -663,11 +663,18 @@ function MoneyRequestConfirmationList(props) {
                 title={props.iouComment}
                 description={translate('common.description')}
                 onPress={() => {
-                    if (props.isEditingSplitBill) {
-                        Navigation.navigate(ROUTES.EDIT_SPLIT_BILL.getRoute(props.reportID, props.reportActionID, CONST.EDIT_REQUEST_FIELD.DESCRIPTION));
+                    if (!props.isEditingSplitBill) {
                         return;
                     }
-                    Navigation.navigate(ROUTES.MONEY_REQUEST_DESCRIPTION.getRoute(props.iouType, props.reportID));
+                    Navigation.navigate(
+                        ROUTES.MONEY_REQUEST_STEP_DESCRIPTION.getRoute(
+                            CONST.IOU.ACTION.EDIT,
+                            CONST.IOU.TYPE.SPLIT,
+                            transaction.transactionID,
+                            props.reportID,
+                            Navigation.getActiveRouteWithoutParams(),
+                        ),
+                    );
```

Comment:
Why we are assuming that this is going to be used for edits only? (similar question for `MoneyTemporaryForRefactorRequestConfirmationList.js`)

Reply:
@s77rt if `props.isEditingSplitBill` is false, we will disable this field by this logic 
https://github.com/Expensify/App/blob/9dbc389ea01748f61f39f4412cbbc804cf306a58/src/components/MoneyRequestConfirmationList.js#L754

Because   ` isReadOnly={!isEditingSplitBill}`

Reply:
The isReadOnly is a different prop and we can't assume it will always be related to isEditingSplitBill.

Reply:
> Why we are assuming that this is going to be used for edits only?

@s77rt  In that time, there was only editing split bill that used this component and this component MoneyRequestConfirmationList also be removed in this refactor phrase. But you're right, we can't assume that this is going to be used for edits only. So I updated in [this commit](https://github.com/Expensify/App/pull/35137/commits/223373475223cc3d412d265f410fb9f45f7be483) to remove condition
```
 if (!props.isEditingSplitBill) {
                        return;
}
```

and using iouType as param of navigate function

Reply:
@DylanDylann Is there any need to change the logic there? Can't we just keep using the same logic (i.e. keep isEditingSplitBill conditions) and navigate to the correct pages?

Reply:
@s77rt We need to change logic to use a new route to `IOURequestStepDescription` for both case. The above code is outdated. This is code after updating

<img width="781" alt="Screenshot 2024-01-31 at 23 51 43" src="https://github.com/Expensify/App/assets/141406735/1821b399-1aea-4e2c-9c4c-8b830c055265">